plugins/webp-uploads/picture-element.php

```diff
@@ -142,7 +145,15 @@ function webp_uploads_wrap_image_in_picture( string $image, string $context, int
 		return false;
 	};
 	add_filter( 'wp_calculate_image_srcset_meta', $filter );
-	$original_image_without_srcset = wp_get_attachment_image( $attachment_id, $original_sizes, false, array( 'src' => $original_image ) );
+	$original_image_without_srcset = wp_get_attachment_image(
+		$attachment_id,
+		$original_sizes,
+		false,
+		array(
+			'src' => $original_image,
+			'alt' => $alt,
+		)
+	);
 	remove_filter( 'wp_calculate_image_srcset_meta', $filter );
```

Comment:
I think there is a better approach here, one critically which will preserve all of the other attributes. The HTML tag processor can be leveraged instead like as follows:

```html
$processor = new WP_HTML_Tag_Processor( $image );
if ( $processor->next_tag( array( 'tag_name' => 'img' ) ) ) {
    $processor->remove_attribute( 'srcset' );
    $image = $processor->get_updated_html();
}
```

Nevertheless, why is the `srcset` being removed in the first place? Shouldn't it be retained for the sake of browsers that don't support `picture`?

Reply:
Yes it will be a  good practice to add srcset to both `<source>` and `<img>` elements. I couldn't find why it is being removed.

On `<source>` elements:
- Provides options for different viewport sizes or device pixel ratios.
- Allows browsers to choose the most appropriate image for the current conditions.

On the `<img>` element:
- Acts as a fallback for browsers that don't support <picture> or <source>.
- Ensures responsiveness even if <source> elements are ignored.

Reply:
So yeah, then this entire block of code can be removed and instead of using `$original_image_without_srcset` we can just use `$image` as was passed into this function originally.